### DISCUSSION TITLE
Controlling elements shown on web ui

### DISCUSSION THREAD

**Discussion by nospam2k** [arendst/Tasmota#22650] [General]
I have made a custom hardware device. I'm a little confused on controlling the template and elements on the web ui. I need to have two buttons show that are only blink buttons. One button will hold gpio6 on for 6 seconds and then turn off. The other will hold gpio 6 only for 1 second and turn off. I made the following rules using "relays" on gpio 0, 1, 6 but that shows 3 buttons. I only want two, but I can't figure out how to use some other "relay" for gpio6 that will switch on and off gpio6. I've made:
```
Rule1 ON POWER1#state=1 DO Backlog POWER3 1; Delay 10; POWER3 0; POWER1 0 ENDON
Rule2 ON POWER2#state=1 DO Backlog POWER3 1; Delay 60; POWER3 0; POWER2 0 ENDON
WebButton1 PC On
WebButton2 PC Off
```
And that works, but I have 3 buttons and they all have ON, OFF over them which I don't need. Soo... how to I get two buttons with no ON, OFF over them to run the rules to power gpio6?

Thanks!

**Comment by sfromis** [comment]
Basically, do not expect to adapt the web UI to detailed ideas about what you like to have. It is deliberately rudimentary, meant for admin/test purposes, with the expectation that "more detailed" control happens via your backend smarthome setup. When you have 3 relays, you do not get to omit any of them from the front page.

IIRC, you can however use the `LedPower` command to get closer to what you want.
https://tasmota.github.io/docs/Lights/#status-leds

OTOH, if your device is using ESP32, you can use the Berry language to have detailed control over pins, and also make your own UI. Rules are pretty basic in what you can do how, but Berry is a full programming language.

Beware that using Delay in rules is not reliable, the interval can be cut short by other things happening, but there is a better way. Instead of `POWER3 1; Delay 60; POWER3 0` you can use `TimedPower3 6000,1` - interval in ms, and it will power off again after that time.

**Comment by nospam2k** [comment]
I appreciate that the interface is rudimentary. I just like the granularity of Tasmota. I wrote my own software and interface, but I'm trying to move all my own custom hardware onto Tasmota for use in HA. I probably won't see the web ui often, it's just me being OCD ;)

My main question is how do I know what shows a web element and what doesn't? Ex. Buttons, switches don't seem to show an web element but relay does. I can't seem to find a break down on the various gpio devices and their relation to the webui or function.

**Reply by Sfre12** [reply]
You can configure your own buttons by using the script language. with this buttons, you can have a different label on the button regarding to the state. (But you must compile your own version with script language support)
```
Button:
bu(vn txt1 txt2) (up to 4 buttons may be defined in one row)
vn = name of variable to hold button state
txt1 = text of ON state of button
txt2 = text of OFF state of button
```
look here:
https://tasmota.github.io/docs/Scripting-Language/#script-sections

**Reply by sfromis** [reply]
The starting point is that none will, but power channels (like relays) and sliders (like lights) will have web UI interaction elements. And sensors show up, but no interactions. Other than that, only a few specialized features like shutters.

As you cannot simulate inputs, having Button/Switch on the web UI would match how things can work, but relays are well-suited to being toggled.
